ansible_base/lib/utils/encryption.py

```diff
@@ -37,34 +38,62 @@ def __init__(self):
         self._encryption_key = self.key[32:]
         self._backend = default_backend()
 
+    def extract_data(self, raw_data: str) -> str:
+        # Sequential data extract for code reuse in is_encrypted_string and decrypt_string
+        raw_data = raw_data[len(ENCRYPTED_STRING) :]
+
+        # If the encrypted string contains a UTF8 marker, discard it
+        if raw_data.startswith('UTF8$'):
+            raw_data = raw_data[len('UTF8$') :]
+
+        # Extract the algorithm and ensure its what we expect
+        algo, data = raw_data.split('$', 1)
+        if algo != ENCRYPTION_METHOD:
+            raise ValueError(f'Unsupported algorithm: {algo}')
+
+        return data
+
+    def is_encrypted_string(self, value: str, invalid_algo_is_fatal: bool) -> bool:
+        # Ensure input is a string already encrypted by our algorithm
+        # by comparing with the decrypted and re-encrypted values
+        if not value.startswith(ENCRYPTED_STRING):
+            return False
+
+        try:
+            data = self.extract_data(value)
+        except ValueError as ve:
+            # value error is raised when algorithm is unsupported
+            if invalid_algo_is_fatal:
+                raise ve
+
```

Comment:
Don't we want to return `False` here if we got a ValueError but `invalid_algo_is_fatal` is `False`? If so, can you write a test case where we detect that not returning causes an error so we can test it before and after the change?